### DISCUSSION TITLE
LunCo Orbital Mechanics

### DISCUSSION THREAD

**Discussion by 0xIonRod** [LunCoSim/lunco-sim#9] [General]
With respect to the Moon mostly we are interested in relative position and orientation of Sun, Earth, Moon at any given point in time.

One of the first applications is to have an accurate sun position in the sim at any point in time, as well as Earth (using kind of LOD, or hacks to adjust position/size to deal with precision error)

A lot could be used from [ivoyager](https://github.com/ivoyager), however it's port is expected to be in no later than[ end of August](https://github.com/orgs/ivoyager/discussions/5)

**Comment by philiplinden** [comment]
Note that 3-body orbital mechanics (Moon+Earth+satellite) are going to be pretty much required for any sort of mission planning. Precise or long-term mission planning should also take into account solar gravity on top of all this.

Luckily there is a _ton_ of literature on these topics that we can lean on.

I can't tell if ivoyager is simulating n-body physics or if it simply advances time with on-rails trajectories.

**Comment by 0xIonRod** [comment]
I'm thinking of using GMAT for n-body physics simulation. It's very advanced, e.g. for different parts of the orbit you can use different numerical integration algorithms, and different models.

I'd prefer not to go into rabbit hole of writing "number crunching" code were it's possible. There are too many nuances.


Ivoyager is a nice basic starting point. It includes Solar System Visualization engine, as well as some n-body simulation.

**Comment by philiplinden** [comment]
Some stepping stones for incorporating orbital mechanics are:

1. Position & orientation of Planets/Sun propagated over time
2. Mission planning e.g. predicting line-of-sight to orbiting satellites and/or terrestrial stations.
3. Collision prediction/avoidance 

I'll look into these (in order) while I learn GMAT or Basilisk

**Reply by 0xIonRod** [reply]
IMHO collision predication is a a HUGE task.

FYI I've met with [Matteo Manzi ](https://www.researchgate.net/profile/Matteo-Manzi) on Gitcoin, he has nice videos about AI/ML/Orbital mechanics like this one [A Robust Bayesian Agent for Optimal Collision Avoidance Manoeuvre Planning](https://www.youtube.com/watch?v=E_T_DBbp_B4).

I suppose at some point in time he could be interested in at least consulting about collision detection/avoidance. And he is in crypto as well, as he is the founder of CrunchDAO
